dhcpv4/client.go

```diff
@@ -194,17 +195,32 @@ func BroadcastSendReceive(sendFd, recvFd int, packet *DHCPv4, readTimeout, write
 		defer conn.Close()
 		conn.SetReadDeadline(time.Now().Add(readTimeout))
 
-		buf := make([]byte, MaxUDPReceivedPacketSize)
-		n, _, _, _, err := conn.(*net.UDPConn).ReadMsgUDP(buf, []byte{})
-		if err != nil {
-			errs <- err
-			return
-		}
-
-		response, err = FromBytes(buf[:n])
-		if err != nil {
-			errs <- err
-			return
+		for {
+			buf := make([]byte, MaxUDPReceivedPacketSize)
+			n, _, _, _, err := conn.(*net.UDPConn).ReadMsgUDP(buf, []byte{})
+			if err != nil {
+				errs <- err
+				return
+			}
+
+			response, err = FromBytes(buf[:n])
+			if err != nil {
+				errs <- err
+				return
+			}
+			// wait for a response message
+			if response.Opcode() != OpcodeBootReply {
+				continue
+			}
+			// if we are not requested to wait for a specific message type,
```

Comment:
why not check the transaction ID?

Reply:
good point. I will check that in addition